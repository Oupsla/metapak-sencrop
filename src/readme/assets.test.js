'use strict';

const assert = require('assert');
const sinon = require('sinon');
const assetsTransformer = require('./assets');

describe('Readme', () => {
  describe('Assets transformer for www configs', () => {
    it('should build the README.md file', done => {
      const fs = {
        readFileAsync: sinon.stub(),
      };
      const PROJECT_DIR = '/lol/';
      const log = {
        error: sinon.stub,
      };

      fs.readFileAsync
        .onFirstCall()
        .returns(Promise.resolve('## Usage\nJust require me\n'));

      assetsTransformer(
        {
          name: 'README.md',
          data: '<!-- something -->\n',
        },
        {
          name: 'module',
          description: 'A great module!',
          devDependencies: {},
          license: 'MIT',
        },
        {
          PROJECT_DIR,
          fs,
          log,
        },
      )
        .then(file => {
          expect(file).toMatchSnapshot();
        })
        .then(done)
        .catch(done);
    });

    it('should build the README.md file with links', done => {
      const fs = {
        readFileAsync: sinon.stub(),
      };
      const PROJECT_DIR = '/lol/';
      const log = {
        error: sinon.stub,
      };

      fs.readFileAsync
        .onFirstCall()
        .returns(Promise.resolve('## Usage\nJust require me\n'));

      assetsTransformer(
        {
          name: 'README.md',
          data: '<!-- something -->\n',
        },
        {
          name: 'module',
          description: 'A great module!',
          metapak: { configs: ['jsarch', 'jsdocs'] },
          devDependencies: {},
          license: 'MIT',
        },
        {
          PROJECT_DIR,
          fs,
          log,
        },
      )
        .then(file => {
          expect(file).toMatchSnapshot();
        })
        .then(done)
        .catch(done);
    });

    it('should let pass other files', () => {
      assert.deepEqual(
        assetsTransformer(
          {
            name: 'YOLO',
            data: 'Carpe diem\n',
          },
          {},
          {},
        ),
        {
          name: 'YOLO',
          data: 'Carpe diem\n',
        },
      );
    });
  });
});
