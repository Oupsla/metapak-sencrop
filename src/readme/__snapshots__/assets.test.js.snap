// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Readme Assets transformer for www configs should build the README.md file 1`] = `
Object {
  "data": "<!-- something -->
# module
> A great module!


## Usage
Just require me

# License
[MIT](https://github.com/sencrop/module/blob/master/LICENSE.md)
",
  "name": "README.md",
}
`;
